zerver/tests/test_docs.py

```diff
@@ -106,17 +107,19 @@ def _test(
 
     def test_api_doc_endpoints(self) -> None:
         current_dir = os.path.dirname(os.path.abspath(__file__))
-        api_docs_dir = os.path.join(current_dir, "..", "..", "templates/zerver/api/")
-        files = os.listdir(api_docs_dir)
-
-        def _filter_func(fp: str) -> bool:
-            ignored_files = ["sidebar_index.md", "index.md", "missing.md"]
-            return fp.endswith(".md") and not fp.startswith(".") and fp not in ignored_files
+        api_docs_file = os.path.join(
+            current_dir, "..", "..", "templates/zerver/help/include/rest-endpoints.md"
+        )
+        api_docs_list = open(api_docs_file, "r").read()
+        ENDPOINT_REGEXP = re.compile(r"api/\s*(.+?)\s*\)")
+        endpoint_list = re.findall(ENDPOINT_REGEXP, api_docs_list)
 
-        files = list(filter(_filter_func, files))
+        docs_file = os.path.join(current_dir, "..", "..", "templates/zerver/api/sidebar_index.md")
+        docs_list = open(docs_file, "r").read()
+        endpoint_list = endpoint_list + re.findall(ENDPOINT_REGEXP, docs_list)
```

Comment:
I think we want to do just do a `client_get` to fetch the actual HTML page, and parse the links on the page.  

(Since we may be trying to auto-generate `rest-endpoints.md` in the future, too)

Reply:
I thought about it, but if we use the client to get the actual raw HTML, then there is no way of distinguishing sidebar links and the main rest-endpoints links. Distinguishing the two kinds of links might make the code messy.

Reply:
I realised that we need to(and we are) test those as well. Changed the code to use the HTML page for parsing links.